docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+filepath: `journal/${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}.md`
+---
+
+<!-- The actual contents of the template begin after the `---` thematic break immediately below this line-->
+---
+# ${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}
+```
+
+Since there is no use of the `${title}` variable, opening the daily note behaves exactly as it does today and automatically opens the note with no further user interaction.
+### Eliminate all `foam.openDailyNote` settings
+
+Now that all of the functionality of the `foam.openDailyNote` settings have been obviated, these settings can be removed:
+
+* `foam.openDailyNote.directory`, `foam.openDailyNote.filenameFormat`, and `foam.openDailyNote.fileExtension` can be specified in the `filepath` metadata of the daily note template.
+* `foam.openDailyNote.titleFormat` has been replaced by the ability to fully template the daily note, including the title.
+
+## Summary: resulting behaviour
+
+### `Foam: Create New Note`
+
+A new command optimized for speedy creation of new notes. This will become the default way to create new notes. In its fastest form, it simply opens the new note with no further user interaction.
+
+### `Foam: Open Daily Note`
+
+Simplified since it no longer has its custom settings, and re-uses all the same implementation code as `Foam: Create New Note`.
+Templates can now be used with daily notes.
+
+### Navigating to missing wikilinks
+
+Now creates the new notes using the default template. Re-uses all the same implementation code as `Foam: Create New Note`
+
+### `Foam: Create Note From Template`
+
+Almost the exact same as it is today. However, with `${title}` and `filepath` templating, users will have less changes to make in the filepath confirmation step.
+It's the slower but more powerful version of `Foam: Create New Note`, allowing you to pick any template, as well as customize he filepath.
+
+## Open questions
+
+### Arbitrary hotkey -> template mappings?
+
+`Foam: Open Daily Note` and `Foam: Create New Note` only differ by their hotkey and their default template setting.
+Is there a reason/opportunity to abstract this further and allow for users to define custom `hotkey -> template` mappings?
+
+```jsonc
+"foam.templateHotkeys": {
+  "default": {
+    "hotkey": "Alt+C"
+    "template": "my_default_template.md"
+  },
+  "daily": {
+    "hotkey": "Alt+D",
+    "template": "my_daily_template.md"
+  },
+  "Fancy Note": {
+    "hotkey": "Alt+E",
+    "template": "my_fancy_note_template.md"
+  }
+}
+```
+
+## Limitations of this proposal
+
+### No `dateFormat`
+
+`foam.openDailyNote.filenameFormat` uses `dateFormat()` to put the current timestamp into the daily notes filename. This is much more flexible than what is available in VSCode Snippet variables. Before daily notes are switched over to use templates, we might have to come up with another mechanism/syntax to allow for calls to `dateFormat()` within template files.
```

Comment:
I agree we'd need this

Reply:
Moved this section from `Limitations` into its own implementation step.